File: samples/java/Hello.java
import java.util.Objects;

public class Hello {
    private final String name;

    public Hello(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Hello(" + name + ")";
    }

    public static void main(String[] args) {
        Hello hello = new Hello("World");
        System.out.println("Hello, " + hello.name + "!");
        System.out.println(Objects.toString(hello));
    }
}

File: samples/rust/hello.rs
use std::fmt;

/// A greeting's name, aliased to exercise a `type` alias.
type Name = String;

/// How loudly to greet.
enum Volume {
    Quiet,
    Loud,
}

struct Greeting {
    name: Name,
    volume: Volume,
}

impl Greeting {
    fn new(name: &str) -> Self {
        Greeting {
            name: name.to_string(),
            volume: Volume::Quiet,
        }
    }

    fn parse(text: &str) -> Self {
        match text.split_once(',') {
            Some((name, _rest)) => Greeting::new(name.trim()),
            None => Greeting::new(text.trim()),
        }
    }

    fn suffix(&self) -> &'static str {
        match self.volume {
            Volume::Quiet => ".",
            Volume::Loud => "!",
        }
    }
}

impl fmt::Display for Greeting {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "Hello, {}{}", self.name, self.suffix())
    }
}

fn main() {
    let mut greeting = Greeting::parse("World, and everyone else");
    greeting.volume = Volume::Loud;
    println!("{}", greeting);
}
